turbinia/workers/hadoop.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task to analyse Hadoop AppRoot files."""
+
+from __future__ import unicode_literals
+
+import codecs
+import os
+import subprocess
+
+from turbinia import TurbiniaException
+
+from turbinia.evidence import ReportText
+from turbinia.lib.utils import extract_artifacts
+from turbinia.workers import TurbiniaTask
+
+
+class HadoopAnalysisTask(TurbiniaTask):
+  """Task to analyse Hadoop AppRoot files."""
+
+  def _AnalyzeHadoopAppRoot(self, collected_artifacts):
+    """Runs a naive AppRoot files parsing method.
+
+    This extracts strings from the saved task file, and searches for usual
+    post-compromise suspicious patterns.
+
+    TODO: properly parse the Proto. Some documentation can be found over there:
+    https://svn.apache.org/repos/asf/hadoop/common/branches/branch-0.23.7/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/proto/yarn_protos.proto
+
+    Args:
+      collected_artifacts(list(str)): a list of paths to extracted files
+    Returns:
+      list(str): the result report, as a list of lines.
+    """
+    report = []
+    strings_report = []
+    evil_commands = []
+    for filepath in collected_artifacts:
+      strings_report.append('Strings for {0:s}:'.format(filepath))
+      proc = subprocess.Popen(
+          'strings -a "{0:s}"'.format(filepath), shell=True,
+          stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+      strings_output, _ = proc.communicate()
+      strings_output = codecs.decode(strings_output, 'utf-8')
+      strings_report.append(strings_output)
+      for line in strings_output.splitlines():
+        if (line.find('curl') > 0) or (line.find('wget') > 0):
```

Comment:
I was all set to approve/merge this, and I realized that I think there is a bug here.  If the string we're searching for is at the start of the line, then it won't include it.  I think we want `>=` here.  Sorry I didn't catch it earlier.